### PR TITLE
Add domnantas lily58 keymap

### PR DESCRIPTION
**Pull request by domnantas**
## Description
Add my keymap for lily58

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/lily58/keymaps/domnantas/keymap.c [hunk 1]
```diff
@@ -0,0 +1,266 @@
+#include QMK_KEYBOARD_H
```
[the remark this answers is not in the document]
Comment (domnantas) [reply keyboards/lily58/keymaps/domnantas/keymap.c:1]: Thanks for review! Added

File: keyboards/lily58/keymaps/domnantas/keymap.c [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+#include <stdio.h>
```
[the remark this answers is not in the document]
Comment (domnantas) [reply keyboards/lily58/keymaps/domnantas/keymap.c:18]: On [Line 250](https://github.com/qmk/qmk_firmware/pull/10910/files#diff-e82b5668f3ef95f8d1fdbb28e76f516e5203eed4e9ddabff1ffe3c6db2a863d7R250) `sprintf` is used. `oled_write_ln(get_current_wpm(), false);` didn't compile. Error thrown:

```
Compiling: keyboards/lily58/keymaps/domnantas/keymap.c                                             keyboards/lily58/keymaps/domnantas/keymap.c: In function ‘render_status’:
keyboards/lily58/keymaps/domnantas/keymap.c:251:19: error: passing argument 1 of ‘oled_write_ln’ makes pointer from integer without a cast [-Werror=int-conversion]
     oled_write_ln(get_current_wpm(), false);
                   ^
In file included from quantum/quantum.h:166:0,
                 from keyboards/lily58/rev1/rev1.h:5,
                 from keyboards/lily58/lily58.h:4,
                 from keyboards/lily58/keymaps/domnantas/keymap.c:17:
drivers/oled/oled_driver.h:214:6: note: expected ‘const char *’ but argument is of type ‘uint8_t {aka unsigned char}’
 void oled_write_ln(const char *data, bool invert);
      ^
cc1: all warnings being treated as errors
```

I tried typecasting like this:
```c
oled_write_ln((char) get_current_wpm(), false)
```
but that didn't work either. I'm not familiar with C enough to figure out what's wrong.

[the remark this answers is not in the document]
Comment (fauxpark) [reply keyboards/lily58/keymaps/domnantas/keymap.c:18]: You'll need to print `get_current_wpm()` into a string first, then pass that to `oled_write_ln()`.

File: keyboards/lily58/keymaps/domnantas/keymap.c [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+     *                   | LGUI | LAlt |LOWER | / Space /       \Enter \  | Bksp |RAISE | Del  |
+     *                   |      |      |      |/       /         \      \ |      |      |      |
+     *                   `----------------------------'           '------''--------------------'
+     */
+    [_QWERTY] = LAYOUT( \
+        KC_ESC,   LT_A_OG, LT_C_CA, LT_E_OG, LT_E_DO, LT_I_OG,                  LT_S_CA, LT_U_OG, LT_U_MA, KC_9,    KC_0,    LT_Z_CA, \
+        KC_TAB,   KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                     KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_MINS, \
+        KC_LSFT,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                     KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, \
+        KC_LCTRL, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,   KC_LBRC, KC_RBRC, KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RALT, \
+                                KC_LGUI, KC_LALT, MO(_LOWER), KC_SPC,  KC_ENT,  KC_BSPC, MO(_RAISE), KC_DEL \
```
Comment (fauxpark) [inline keyboards/lily58/keymaps/domnantas/keymap.c:90]:
```suggestion
    [_QWERTY] = LAYOUT(
        KC_ESC,   LT_A_OG, LT_C_CA, LT_E_OG, LT_E_DO, LT_I_OG,                  LT_S_CA, LT_U_OG, LT_U_MA, KC_9,    KC_0,    LT_Z_CA,
        KC_TAB,   KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                     KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_MINS,
        KC_LSFT,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                     KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
        KC_LCTRL, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,   KC_LBRC, KC_RBRC, KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RALT,
                                KC_LGUI, KC_LALT, MO(_LOWER), KC_SPC,  KC_ENT,  KC_BSPC, MO(_RAISE), KC_DEL
```

File: keyboards/lily58/keymaps/domnantas/keymap.c [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+     *                   | LGUI | LAlt |LOWER | / Space /       \ Enter\  | Bksp |RAISE | Del  |
+     *                   |      |      ||||||||/       /         \      \ |      |      |      |
+     *                   `----------------------------'           '------''--------------------'
+     */
+    [_LOWER] = LAYOUT( \
+        _______, LT_1,    LT_2,    LT_3,    LT_4,    LT_5,                      LT_6,    LT_7,    LT_8,    LT_9,    LT_0,    KC_GRV,\
+        _______, LT_4,    LT_5,    LT_6,    _______, _______,                   _______, _______, _______, _______, _______, _______, \
+        _______, LT_7,    LT_8,    LT_9,    LT_0,    _______,                   _______, _______, _______, _______, _______, _______, \
+        _______, LT_EQL, KC_PPLS, KC_PMNS, KC_PIPE, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+                                   _______, _______, _______, _______, _______, _______, _______, _______\
```
Comment (fauxpark) [inline keyboards/lily58/keymaps/domnantas/keymap.c:112]:
```suggestion
    [_LOWER] = LAYOUT(
        _______, LT_1,    LT_2,    LT_3,    LT_4,    LT_5,                      LT_6,    LT_7,    LT_8,    LT_9,    LT_0,    KC_GRV,
        _______, LT_4,    LT_5,    LT_6,    _______, _______,                   _______, _______, _______, _______, _______, _______,
        _______, LT_7,    LT_8,    LT_9,    LT_0,    _______,                   _______, _______, _______, _______, _______, _______,
        _______, LT_EQL, KC_PPLS, KC_PMNS, KC_PIPE, _______, _______, _______, _______, _______, _______, _______, _______, _______,
                                   _______, _______, _______, _______, _______, _______, _______, _______
```

File: keyboards/lily58/keymaps/domnantas/keymap.c [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+     *                   | LGUI | LAlt |LOWER | / Space /       \ Enter\  | Bksp |RAISE | Del  |
+     *                   |      |      |      |/       /         \      \ |      ||||||||      |
+     *                   `----------------------------'           '------''--------------------'
+     */
+    [_RAISE] = LAYOUT( \
+        _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,                     KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_PSCR, \
+        _______, KC_F11,  KC_F12,  _______, _______, _______,                   _______, KC_PGUP, KC_HOME, KC_UP,   KC_END,  KC_VOLU, \
+        _______, _______, _______, _______, _______, _______,                   _______, KC_PGDN, KC_LEFT, KC_DOWN, KC_RGHT, KC_VOLD, \
+        _______, C(KC_Z), C(KC_X), C(KC_C), C(KC_V), _______, _______, _______, _______, _______, KC_MPRV, KC_MNXT, KC_MPLY, _______, \
+                                   _______, _______, _______, _______, _______, _______, _______, _______ \
```
Comment (fauxpark) [inline keyboards/lily58/keymaps/domnantas/keymap.c:134]:
```suggestion
    [_RAISE] = LAYOUT(
        _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,                     KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_PSCR,
        _______, KC_F11,  KC_F12,  _______, _______, _______,                   _______, KC_PGUP, KC_HOME, KC_UP,   KC_END,  KC_VOLU,
        _______, _______, _______, _______, _______, _______,                   _______, KC_PGDN, KC_LEFT, KC_DOWN, KC_RGHT, KC_VOLD,
        _______, C(KC_Z), C(KC_X), C(KC_C), C(KC_V), _______, _______, _______, _______, _______, KC_MPRV, KC_MNXT, KC_MPLY, _______,
                                   _______, _______, _______, _______, _______, _______, _______, _______
```

File: keyboards/lily58/keymaps/domnantas/keymap.c [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+     *                   | LGUI | LAlt |LOWER | /Space  /       \Enter \  | Bksp |RAISE | Del  |
+     *                   |      |      ||||||||/       /         \      \ |      ||||||||      |
+     *                   `----------------------------'           '------''--------------------'
+     */
+    [_ADJUST] = LAYOUT( \
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, \
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, \
+        KC_CAPS, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, \
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,\
+                                   _______, _______, _______, _______, _______,  _______, _______, _______ \
```
Comment (fauxpark) [inline keyboards/lily58/keymaps/domnantas/keymap.c:156]:
```suggestion
    [_ADJUST] = LAYOUT(
        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
        KC_CAPS, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
                                   _______, _______, _______, _______, _______,  _______, _______, _______
```

File: keyboards/lily58/keymaps/domnantas/rules.mk [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/lily58/keymaps/domnantas/rules.mk:2]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by domnantas** [comment]
@fauxpark Thanks for the review!

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thank you for the contribution!!

**Pull request merged**